merfi/backends/rpm_sign.py

```diff
@@ -39,6 +39,12 @@ def clear_sign(self, path, command):
         """
         logger.info('signing: %s' % path)
         out, err, code = util.run_output(command)
+        if code != 0:
+            logger.error('rpm-sign non-zero exit code %d' % code)
+            raise RuntimeError('stdout: %s stderr: %s' % (out, err))
```

Comment:
this will need a bit of re-arranging, because stdout and stderr might be several lines, not just one tiny bit, so in this way it would be too hard to read. I suggest logging them separately and then raise at the end. Something like (pseudo code):

    for line in stderr.split('\n'):
        logger.error('stderr: %s', line)
    for line in stdout.split('\n'):
        logger.error('stdout: %s', line)
    raise RuntimeError('rpm-sign non-zero exit code %d', code)

Reply:
good catch. I've amended the change to avoid logging stdout too when we know it's blank.